src/inmanta/protocol/openapi/model.py

```diff
@@ -68,6 +70,61 @@ class SchemaBase(BaseModel):
     anyOf: Optional[Sequence["SchemaBase"]] = None
     enum: Optional[List[str]] = None
 
+    def from_dict(self: S, schema: Dict[str, Any]) -> S:
+        """
+        Build an object with the same type as this object
+
+        :param schema: The schema of the object to build
+        """
+        return SchemaBase(**schema)
+
+    def resolve(self: S, ref_prefix: str, known_schemas: Dict[str, Any]) -> S:
+        """
+        Returns this object or the one this object is refering to.
+
+        :param ref_prefix: The prefix this object ref should have if it is only a reference
+        :param known_schemas: A dict of known schemas, the keys are the reference diminished from their prefix
+
+        :raises ValueError: If the schema has a badly formed ref
+        """
+        if not self.ref:
+            return self
+
+        if not self.ref.startswith(ref_prefix):
+            raise ValueError(f"Schema reference (={self.ref}) doesn't start with the expected prefix: '{ref_prefix}'")
+
+        reference = self.ref[len(ref_prefix) :]
+        return self.from_dict(known_schemas[reference])
+
+    def recursive_resolve(
+        self: S, ref_prefix: str, known_schemas: Dict[str, Any], update: Dict[str, Any], deep: bool = True
+    ) -> S:
+        """
+        Returns this object of the one this object is refering to, and resolve all the nested schema it contains.
+
+        :param ref_prefix: The prefix this object ref should have if it is only a reference
+        :param known_schemas: A dict of known schemas, the keys are the reference diminished from their prefix
+        :param update: A dict of attributes of this object to update
```

Comment:
This argument I don't understand

Reply:
Better now? https://github.com/inmanta/inmanta-core/pull/3217/commits/d5d45388d854279635bfe30013df9c33c075814a